MaterialLib/Fluid/Density/IdealGasLaw.h

```diff
@@ -0,0 +1,101 @@
+/*!
+   \file  IdealGasLow.h
+   \brief Declaration of class IdealGasLow for fluid density by the ideal gas
+          law depending on one variable linearly.
+
+   \author Wenqing Wang
+   \date Jan 2015
+
+   \copyright
+    Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+               Distributed under a Modified BSD License.
+               See accompanying file LICENSE.txt or
+               http://www.opengeosys.org/project/license
+ */
+#ifndef IDEAL_GAS_LAW_H_
+#define IDEAL_GAS_LAW_H_
+
+#include <string>
+#include <cassert>
+
+#include "MaterialLib/Fluid/FluidProperty.h"
+#include "MaterialLib/PhysicalConstant.h"
+
+namespace MaterialLib
+{
+namespace Fluid
+{
+/// Fluid density by ideal gas law
+class IdealGasLaw final : public FluidProperty
+{
+public:
+    ///   \param molar_mass Molar mass of the gas phase.
+    explicit IdealGasLaw(const double molar_mass)
+    : FluidProperty(),_molar_mass(molar_mass)
+    {
+    }
+
+    /// Get density model name.
+    std::string getName() const override
+    {
+        return "Ideal gas law";
+    }
+
+    /// Get density value.
+    /// \param var_vals Variable values in an array. The order of its elements
+    ///                 is given in enum class PropertyVariableType.
+    double getValue(const ArrayType& var_vals) const override
+    {
+        return _molar_mass * var_vals[static_cast<int> (PropertyVariableType::pg)] /
+               (PhysicalConstant::IdealGasConstant *
+                var_vals[static_cast<int> (PropertyVariableType::T)]);
+    }
+
+    /// Get the partial differential of the density with respect to temperature
+    /// or gas pressure.
+    /// \param var_vals  Variable values  in an array. The order of its elements
+    ///                   is given in enum class PropertyVariableType.
+    /// \param var       Variable type.
+    double getdValue(const ArrayType& var_vals,
+            const PropertyVariableType var) const override
+    {
+        assert(var == PropertyVariableType::T || var == PropertyVariableType::pg);
+
+        const int func_id = (var == PropertyVariableType::T) ? 0 : 1;
```

Comment:
Is it not possible to call `dIdealGasLaw_dT` or `dIdealGasLaw_dp`  using an `if` construct? Why do you need function pointers?